code/game/objects/structures/syndicate_uplink_beacon.dm

```diff
@@ -0,0 +1,115 @@
+/// Device that traitors can craft in order to be sent a new, undisguised uplink
+/obj/structure/syndicate_uplink_beacon
+	name = "suspicious beacon"
+	icon = 'icons/obj/machines/telecomms.dmi'
+	icon_state = "relay_traitor"
+	desc = "This ramshackle device seems capable of recieving and sending signals for some nefarious purpose."
+	density = TRUE
+	anchored = TRUE
+	/// Traitor's code that they speak into the radio
+	var/uplink_code
+	/// weakref to person who is going to use the beacon to get a replacement uplink
+	var/datum/weakref/owner
+	/// while constructed the teleport beacon is still active
+	var/obj/item/beacon/teleport_beacon
+	/// Radio that the device needs to listen to the codeword from the traitor
+	var/obj/item/radio/listening_radio
+	/// prevents traitor from activating teleport_beacon proc too much in a small period of time
+	COOLDOWN_DECLARE(beacon_cooldown)
+
+/obj/structure/syndicate_uplink_beacon/Initialize(mapload)
+	. = ..()
+	register_context()
+
+	var/static/list/tool_behaviors = list(
+		TOOL_SCREWDRIVER = list(
+			SCREENTIP_CONTEXT_RMB = "Deconstruct",
+		),
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_behaviors)
+	listening_radio = new(src)
+	listening_radio.canhear_range = 0
+	teleport_beacon = new(src)
+
+/obj/structure/syndicate_uplink_beacon/attack_hand(mob/living/user, list/modifiers)
+	if(!IS_TRAITOR(user))
+		balloon_alert(user, "don't know how to use!")
+		return
+	if(IS_WEAKREF_OF(owner, user))
+		balloon_alert(user, "already synchronized to you!")
+		return
+	if(owner != null)
+		balloon_alert(user, "already claimed!")
+		return
+	var/datum/looping_sound/typing/typing_sounds = new(src, /*start_immediately =*/TRUE)
```

Comment:
```suggestion
	var/datum/looping_sound/typing/typing_sounds = new(src, start_immediately = TRUE)
```

99% sure this is what mothblocks meant, unless `start_immediately` isn't the named arg, in which case you should change it to be the real named arg instead.

Reply:
or does `new()` not accept named args? ignore me if this doesn't compile

Reply:
it complied, i think that new lets it through if its a datum but not if its an atom so i did add commented named arguments to the replacement beacon which i forgot to do! oops